core/src/main/java/org/elasticsearch/action/bulk/BulkItemResponse.java

```diff
@@ -40,7 +44,14 @@
     /**
      * Represents a failure.
      */
-    public static class Failure {
+    public static class Failure implements Writeable<Failure>, ToXContent {
+        static final String INDEX_FIELD = "index";
+        static final String TYPE_FIELD = "type";
+        static final String ID_FIELD = "id";
+        static final String CAUSE_FIELD = "cause";
+        static final String STATUS_FIELD = "status";
+
+        public static final Failure PROTOTYPE = new Failure(null, null, null, null);
```

Comment:
I think it is a side effect of Writeable's contact. I don't like it either but a backport isn't the place to remove it because it'll stick around in master. I've filed https://github.com/elastic/elasticsearch/issues/17085 so we can talk about it.